pgjdbc/src/main/java/org/postgresql/jdbc/PgArray.java

```diff
@@ -876,11 +881,16 @@ public ResultSet getResultSetImpl(long index, int count, Map<String, Class<?>> m
   public String toString() {
     if (fieldString == null && fieldBytes != null) {
       try {
-        Object array = readBinaryArray(1,0);
-        java.sql.Array tmpArray = connection.createArrayOf(getBaseTypeName(), (Object[]) array);
-        fieldString = tmpArray.toString();
+        Object array = readBinaryArray(1, 0);
+
+        if (PrimitiveArraySupport.isSupportedPrimitiveArray(array)) {
+          fieldString = PrimitiveArraySupport.arrayToString(connection.getTypeInfo().getArrayDelimiter(oid), array);
```

Comment:
Done.
The static utility arrayToString method was used more before binary support was added. I did not realize it had dropped to only 1 use when binary representation became preferred.